libsolidity/codegen/CompilerContext.cpp

```diff
@@ -167,6 +167,18 @@ CompilerContext& CompilerContext::appendJump(eth::AssemblyItem::JumpType _jumpTy
 	return *this << item;
 }
 
+CompilerContext& CompilerContext::appendInvalid()
+{
+	return *this << Instruction::INVALID;
+}
+
+CompilerContext& CompilerContext::appendConditionalInvalid()
+{
+	*this << Instruction::ISZERO;
+	eth::AssemblyItem afterTag = appendConditionalJump();
+	return *this << Instruction::INVALID << afterTag;
```

Comment:
Thanks! I tried the following:

```c++
	unsigned startStackHeight = stackHeight();
	*this << Instruction::ISZERO;
	eth::AssemblyItem afterTag = appendConditionalJump();
	*this << Instruction::INVALID << afterTag;
	setStackOffset(startStackHeight - 1);
	return *this;
```

However, this didn't seem to do the trick. Were you thinking of something else?